render_math/math.py

```diff
@@ -295,29 +295,38 @@ def pelican_init(pelicanobj):
     if mathjax_settings['process_summary']:
         process_summary.mathjax_script = mathjax_script
 
-def rst_add_mathjax(article):
+def rst_add_mathjax(content):
     """Adds mathjax script for RST"""
 
     # This is only value for .rst files
-    _, ext = os.path.splitext(os.path.basename(article.source_path))
+    _, ext = os.path.splitext(os.path.basename(content.source_path))
     if ext != '.rst':
         return
 
     # If math class is present in text, add the javascript
-    if 'class="math"' in article._content:
-        article._content += "<script type='text/javascript'>%s</script>" % rst_add_mathjax.mathjax_script
+    if 'class="math"' in content._content:
+        content._content += "<script type='text/javascript'>%s</script>" % rst_add_mathjax.mathjax_script
 
-def parse_articles(article_generator):
-    """Adds mathjax script to RST and processes summaries"""
-
-    for article in article_generator.articles:
-        
-        rst_add_mathjax(article)
-
-        if process_summary.mathjax_script is not None:
-            process_summary(article)
+def massage_content(content_generators):
```

Comment:
I don't like the name massage content - it seems too general to me. Also (and this may just be me), the description does not really describe what is being done properly (or it can be described better).